test/unit/block/sharedSharing/src/BlockSharedMemSharing.cpp

```diff
@@ -0,0 +1,154 @@
+/* Copyright 2022 Jeffrey Kelling
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/acc/Traits.hpp>
+#include <alpaka/atomic/Traits.hpp>
+#include <alpaka/block/shared/dyn/Traits.hpp>
+#include <alpaka/block/shared/st/Traits.hpp>
+#include <alpaka/block/sync/Traits.hpp>
+#include <alpaka/test/Array.hpp>
+#include <alpaka/test/KernelExecutionFixture.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+#include <alpaka/workdiv/Traits.hpp>
+#include <alpaka/workdiv/WorkDivMembers.hpp>
+
+#include <catch2/catch.hpp>
+
+//! This tests checks if block-shared memory is shared correctly: only between all threads in a block.
+//!
+//! The check is done by each thread atomically adding 1 to a variable `shared`. Possible outcomes are:
+//! * `shared < blockThreadCount`: memory is not shared between all threads of a block; usually `shared == 1`: memory
+//!   is thread-private (fail)
+//! * `shared > blockThreadCount`: memory is shared between blocks (fail)
+//! * `shared == blockThreadCount`: memory is shared correctly (pass)
+template<typename TAcc>
+ALPAKA_FN_ACC void blockSharedMemSharingTestKernelHelper(TAcc const& acc, std::uint32_t* sums, std::uint32_t& shared)
+{
+    auto const blockThreadIdx(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u]);
```

Comment:
This must be 

```C++
auto const blockThreadIdx = alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u];
auto const gridBlockIdx = alpaka::getIdx<alpaka::Grid, alpaka::Blocks>(acc)[0u];
```

seee: https://github.com/alpaka-group/alpaka/pull/1404

Reply:
m( copied this from the `vectorAdd` but replaced `TIdx` by auto, not noticing, that this looses the type. ... not sure if I want to know why this even works in most compilers ... must be interpreted as one of the ~20 ways in C++ one should rather not use to initialize things... . Thanks.